core/src/test/java/org/quiltmc/mapping/parser/QuiltMappingParserTest.java

```diff
@@ -40,9 +43,62 @@ public class QuiltMappingParserTest {
 			UnpickEntry.UNPICK_MAPPING_TYPE,
 			TransitiveEntry.TRANSITIVE_MAPPING_TYPE);
 
-	public static void main(String[] args) throws IOException {
-		String input = new String(QuiltMappingParserTest.class.getClassLoader().getResourceAsStream("org/quiltmc/mapping/parser/TestMapping.quiltmapping").readAllBytes());
-		System.out.println(input);
-		System.out.println(new QuiltMappingParser(input, TYPES).parse());
+	@Test
+	void main() throws IOException {
+		// test: test mapping file
+		// this test should parse as normal with no issues
+		String testMapping = getInput("org/quiltmc/mapping/parser/TestMapping.quiltmapping");
+
+		System.out.println(testMapping);
+		System.out.println(new QuiltMappingParser(testMapping, TYPES).parse());
+
+		// test: test mapping file with negative argument index
+		// this test should throw an exception
+		String negativeArgIndexTest = getInput("org/quiltmc/mapping/parser/fail_cases/NegativeArgumentIndexTestMapping.quiltmapping");
+
+		try {
+			new QuiltMappingParser(negativeArgIndexTest, TYPES).parse();
+			Assertions.fail("expected a parsing exception in negative arg index test");
+		} catch (Exception ignored) {
+			// this is what we want!
+		}
```

Comment:
For these, use `assertThrows`, since that is clearer what the intedend goal is. This makes it look like we are ignoring the errors from a quick glance.

Reply:
will fix tonight!